From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "benh@amazon.com" <benh@amazon.com>
Date: Tue, 18 May 2021 13:17:30 +1000
Subject: [PATCH] glibc-rh827510.patch

# Upstream BZ 14247# Upstream BZ 14247

2012-06-14  Jeff Law  <law@redhat.com>

	* locale/loadlocale.c (_nl_load_locale): Delay setting
	file->decided until we have successfully loaded the file's
	data.

diff --git a/locale/loadlocale.c b/locale/loadlocale.c
index 9bca30463e39a24ab08cf361fcf8da33e6119aac..e562d993c6bb8f76986dcc15a148597d0224352a 100644
--- a/locale/loadlocale.c
+++ b/locale/loadlocale.c
@@ -170,7 +170,6 @@ _nl_load_locale (struct loaded_l10nfile *file, int category)
   int save_err;
   int alloc = ld_mapped;
 
-  file->decided = 1;
   file->data = NULL;
 
   fd = open_not_cancel_2 (file->filename, O_RDONLY | O_CLOEXEC);
@@ -279,6 +278,7 @@ _nl_load_locale (struct loaded_l10nfile *file, int category)
   newdata->alloc = alloc;
 
   file->data = newdata;
+  file->decided = 1;
 }
 
 void
